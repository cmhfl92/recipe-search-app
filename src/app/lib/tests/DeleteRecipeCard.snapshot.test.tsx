import { render, screen, fireEvent } from '@testing-library/react';
import React from 'react';

const mockDelete = jest.fn();

const DeleteButtonTest = ({ recipeId }: { recipeId: string }) => {
  const handleDeleteRecipe = async (id: string) => {
    mockDelete(recipeId);
  };

  return <button onClick={() => handleDeleteRecipe(recipeId)}>Delete</button>;
};

describe('Delete Button', () => {
  it('deletes recipe card with recipeId', () => {
    const { asFragment } = render(<DeleteButtonTest recipeId='recipeIdTest' />);
    expect(asFragment()).toMatchSnapshot();
  });
});
