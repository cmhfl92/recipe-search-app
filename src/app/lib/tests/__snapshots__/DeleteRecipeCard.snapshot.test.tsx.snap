// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delete Button deletes recipe card with recipeId 1`] = `
<DocumentFragment>
  <button>
    Delete
  </button>
</DocumentFragment>
`;
